packages/ember-debug/tests/main_test.js

```diff
@@ -105,14 +105,12 @@ QUnit.test('Ember.deprecate throws deprecation if second argument is falsy', fun
   });
 });
 
-QUnit.test('Ember.deprecate does not throw deprecation if second argument is a function and it returns true', function() {
+QUnit.test('Ember.deprecate throws deprecation if second argument is a function and it returns true', function() {
   expect(1);
 
-  Ember.deprecate('Deprecation is thrown', function() {
-    return true;
-  }, { id: 'test', until: 'forever' });
-
-  ok(true, 'deprecation was not thrown');
+  throws(() => {
```

Comment:
It works fine though when moving Ember.deprecate out of expectDeprecation callback or using throws. I think it won't be possible with using expectDeprecation. I can't debug it, I don't know how that helper works. Seems like triggering deprecation from inside of Ember.deprecate is too much for it. In other places in tests in this file throws is used. Can I stick to using throws, or should I go with registerHandler approach? @rwjblue